api/src/main/java/io/strimzi/api/kafka/model/KafkaUserSpec.java

```diff
@@ -26,7 +26,14 @@
     private KafkaUserQuotas quotas;
     private KafkaUserTemplate template;
 
-    @Description("Authentication mechanism enabled for this Kafka user.")
+    @Description("Authentication mechanism enabled for this Kafka user. " +
+            "Currently the supported authentication mechanisms are `scram-sha-512`, `tls`, and `tls-noop`. \n\n" +
+            "* `scram-sha-512` will generate secret with SASL SCRAM-SHA-512 credentials.\n" +
+            "* `tls` will generate secret with user certificate for mutual TLS authentication.\n" +
+            "* `tls-noop` will not generate the user certificate but will prepare the user for using mutual TLS authentication with a user certificate generated outside the User Operator.\n" +
```

Comment:
I wonder if `external-tls` might be a better name.

Reply:
Would `tls-external` be better? It aligns better with `tls` as the main part there I think.

Reply:
I can live with that.